src/main/java/helpers/JsonParser.java

```diff
@@ -0,0 +1,35 @@
+package helpers;
+
+import com.coherent.token.ReadTokenManager;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.*;
+
+public class JsonParser {
+//private String token;
+
+//    public void writeToFile(ReadTokenManager token) {
+//        try (FileWriter writer = new FileWriter("src/main/resources/" + cart.getCartName() + ".json")) {
+//            writer.write(gson.toJson(cart));
+//        } catch (IOException e) {
+//            e.printStackTrace();
+//        }
+//    }
+
+
+    public String readFromJson (String response) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonNode = null;
+        try {
+            jsonNode = objectMapper.readTree(response);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        String token = jsonNode.get("access_token").asText();
+
+        return token;
+    }
```

Comment:
Could you please refactor this one and use objectMapper.readValue(json, TokenResponse.class)
You will need to create TokenResponse class which will have fields from the response that you get when send request to get token.
json - it is your response from backend. You are getting it in WriteTokenReceiver on line 35 : responseBody = EntityUtils.toString(response.getEntity());

Reply:
Refactored as discussed on a meeting. I deleted the whole JsonParser class and added deserialization logic to the ReadTokenReceiver and WriteTokenReceiver class. I used ObjectMapper class for that.